php/libraries/Database.class.inc

```diff
@@ -1159,5 +1159,83 @@ class Database
         }
     }
 
+    /**
```

Comment:
@driusan The transaction stuff is required because this uploader could add millions of rows and committing on each one of those would take days to complete. The previous way of doing this was to create bulk inserts but it was doing it by writing in the query instead of using params for the prepare statement. 

The choice is a) merge this with the transaction support. b) close this and keep the params in the query (security vulnerability).